src/mpi/request/waitany.c

```diff
@@ -147,6 +147,12 @@ int MPI_Waitany(int count, MPI_Request array_of_requests[], int *indx, MPI_Statu
         }
     }
 
+    if (unlikely(last_disabled_anysource != -1)) {
```

Comment:
This replicates the handling in Waitsome. My thought is that since Testany is similar as one iteration while loop of waitany and if there is such a failure testany will report it the same way.

Reply:
Okay, so I guess you wanted to make a progress at least once, but since this is kind of an error handling case, you don't have to wait, so testany is fine, right?

Reply:
Yes even in error case we want to make one pass of progress, so testany does just that